noiseutil/boring_test.go

```diff
@@ -0,0 +1,37 @@
+//go:build boringcrypto
+// +build boringcrypto
+
+package noiseutil
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Ensure NewGCMTLS validates the nonce is non-repeating
+func TestNewGCMTLS(t *testing.T) {
+	// Test Case 16 from http://csrc.nist.gov/groups/ST/toolkit/BCM/documents/proposedmodes/gcm/gcm-spec.pdf
```

Comment:
This link is now a redirect to https://csrc.nist.gov/projects/block-cipher-techniques/bcm, which eventually links to https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf for GCM. But I didn't see any test cases listed therein – is there a way I see the case referenced here?